### PR TITLE
Adding expression definition

### PR DESCRIPTION
**Pull request by tsurdilo**

### CODE REVIEW

**Comment by tsurdilo** [comment]
related to issue https://github.com/cncf/wg-serverless/issues/87

**Comment by tsurdilo** [comment]
@salaboy @mswiderski can you guys please review this pr? Thanks!

**Comment by salaboy** [comment]
@tsurdilo this is a good addition to make sure that the language support language independent expression evaluators.. 
/LGTM

**Comment by tsurdilo** [comment]
@cathyhongzhang Cathy can you please also have a look? Thanks

**Comment by manuelstein** [comment]
IIUC the eventExpression is meant to be a boolean condition to whether the event should actually cause the trigger to fire. I guess "condition" fits better than "eventExpression" but maybe that's just me.

**Comment by salaboy** [comment]
@manuelstein I tend to agree, so we should have `expressionLanguage` which is kinda correct and `condition` which is where the expressions are defined.  @tsurdilo what do you think about that?

File: workflow/spec/spec.md [hunk 1]
```diff
@@ -375,6 +380,41 @@ should get triggered after this event completes.
...
+
+
+<details><summary><strong>Click to view JSON Schema</strong></summary>
+
+```json
+{
+  "type": "object",
+  "description": "Defines the language and body of expression.",
+  "properties": {
+    "language": {
```
Comment (cathyhongzhang) [inline workflow/spec/spec.md]: Does this "language" refer to the "expressionLanguage"? If so, it needs to be changed.

File: workflow/spec/spec.md [hunk 2]
```diff
@@ -375,6 +380,41 @@ should get triggered after this event completes.
...
+      "type": "string",
+      "description": "Expression language. For example 'spel', 'jexl', 'cel', etc"
```
Comment (cathyhongzhang) [inline workflow/spec/spec.md]: Add "json" as an example.

File: workflow/spec/spec.md [hunk 3]
```diff
@@ -375,6 +380,41 @@ should get triggered after this event completes.
...
+    },
+    "body": {
+      "type": "string",
+      "description": "The expression body"
+    }
```
Comment (cathyhongzhang) [inline workflow/spec/spec.md]: Add "For example, (event1 or event2) and event3"

File: workflow/spec/spec.md [hunk 4]
```diff
@@ -319,7 +324,7 @@ Event state can hold one or more events definitions, so let's define those:
 
 | Parameter | Description | Type | Required |
 | --- | --- | --- | --- |
-| eventExpression |Expression used to associate triggerDefs with event state  | string | yes |
+| [eventExpression](#Expression-Definition) |Expression used to associate triggerDefs with event state  | object | yes |
```
Comment (cathyhongzhang) [inline workflow/spec/spec.md]: Suggest to change to "Expression consisting of Boolean operation of events that will trigger the event state"

File: workflow/spec/spec.md (hunk 2 above)
Comment (tsurdilo) [reply workflow/spec/spec.md]: json is not an expression language, so not sure why?

File: workflow/spec/spec.md (hunk 1 above)
Comment (tsurdilo) [reply workflow/spec/spec.md]: fixed

File: workflow/spec/spec.md (hunk 3 above)
Comment (tsurdilo) [reply workflow/spec/spec.md]: fixed

File: workflow/spec/spec.md (hunk 4 above)
Comment (tsurdilo) [reply workflow/spec/spec.md]: fixed

**Comment by tsurdilo** [comment]
@manuelstein updated names according to your suggestions.

File: workflow/spec/spec.md (hunk 3 above)
Comment (tsurdilo) [reply workflow/spec/spec.md]: added

**Pull request merged**